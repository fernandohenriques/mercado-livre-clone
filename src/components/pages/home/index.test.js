import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { render } from '@testing-library/react';

import Home from './index';

describe('pages <Home />', () => {
  const mockStore = configureMockStore();

  it('Snapshot testing without loading', () => {
    const store = mockStore({ search: { loading: false } });

    const { asFragment } = render(
      <Provider store={store}>
        <Home />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Snapshot testing with loading', () => {
    const store = mockStore({ search: { loading: true } });

    const { asFragment } = render(
      <Provider store={store}>
        <Home />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
