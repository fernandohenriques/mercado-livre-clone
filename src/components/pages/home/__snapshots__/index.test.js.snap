// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages <Home /> Snapshot testing with loading 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="logo"
        role="button"
        tabindex="0"
      >
        <figure>
          <img
            alt="Mercado Livre - Onde comprar e vender de Tudo"
            src="logo.png"
            title="Mercado Livre - Onde comprar e vender de Tudo"
          />
        </figure>
      </div>
      <div
        class="container"
      >
        <input
          class="input input"
          placeholder="Buscar produtos..."
          type="text"
          value=""
        />
        <button
          type="button"
        >
          <img
            alt="Buscar"
            src="ic-search.png"
            title="Buscar"
          />
        </button>
      </div>
    </div>
  </header>
  <main>
    <div
      class="container"
    >
      <div
        class="loading"
      />
    </div>
  </main>
</DocumentFragment>
`;

exports[`pages <Home /> Snapshot testing without loading 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="logo"
        role="button"
        tabindex="0"
      >
        <figure>
          <img
            alt="Mercado Livre - Onde comprar e vender de Tudo"
            src="logo.png"
            title="Mercado Livre - Onde comprar e vender de Tudo"
          />
        </figure>
      </div>
      <div
        class="container"
      >
        <input
          class="input input"
          placeholder="Buscar produtos..."
          type="text"
          value=""
        />
        <button
          type="button"
        >
          <img
            alt="Buscar"
            src="ic-search.png"
            title="Buscar"
          />
        </button>
      </div>
    </div>
  </header>
  <main>
    <div
      class="container"
    >
      <h3>
        Encontre tudo que precisa.
      </h3>
    </div>
  </main>
</DocumentFragment>
`;
